packages/mobile/src/screens/sign-on-screen/components/WelcomeDrawer.tsx

```diff
@@ -0,0 +1,96 @@
+import { fillString, welcomeModalMessages as messages } from '@audius/common'
+import { css } from '@emotion/native'
+import { getNameField } from 'audius-client/src/common/store/pages/signon/selectors'
+import { useDispatch, useSelector } from 'react-redux'
+
+import {
+  Flex,
+  IconArrowRight,
+  IconCloudUpload,
+  Text
+} from '@audius/harmony-native'
+import { Button } from 'app/components/core'
+import { NativeDrawer } from 'app/components/drawer'
+import { useNavigation } from 'app/hooks/useNavigation'
+import { setVisibility } from 'app/store/drawers/slice'
+
+import {
+  ReadOnlyCoverPhotoBanner,
+  ReadOnlyProfilePicture
+} from './AccountHeader'
+
+export const WelcomeDrawer = () => {
+  const { value: displayName } = useSelector(getNameField)
+
+  const navigation = useNavigation()
+
+  const dispatch = useDispatch()
+  const closeModal = () => {
```

Comment:
use the "useModal" / "useDrawer" hook to get all this, i forget what it's called exactly